WcaOnRails/spec/controllers/registrations_controller_spec.rb

```diff
@@ -169,16 +167,29 @@
       expect(registration.competition_id).to eq competition.id
     end
 
+    it "can re-create registration after it was deleted" do
+      registration = FactoryGirl.create :registration, :accepted, :deleted, competition: competition, user_id: user.id
+      registration_competition_event = registration.registration_competition_events.first
+
+      patch :update, id: registration.id, registration: { registration_competition_events_attributes: [ {id: registration_competition_event.id, registration_id: registration.id, competition_event_id: threes_comp_event.id, _destroy: 0} ],
+                                                          comments: "Registered again" }
+      expect(registration.reload.comments).to eq "Registered again"
+      expect(registration.reload.pending?).to eq true
+      expect(registration.reload.accepted?).to eq false
+      expect(registration.reload.deleted?).to eq false
+      expect(flash[:success]).to eq "Updated registration"
+      expect(response).to redirect_to competition_register_path(competition)
+    end
+
     it "can delete registration when on waitlist" do
       registration = FactoryGirl.create :registration, :pending, competition: competition, user_id: user.id
 
       expect(RegistrationsMailer).to receive(:notify_organizers_of_deleted_registration).and_call_original
-      expect do
-        delete :destroy, id: registration.id, user_is_deleting_theirself: true
-      end.to change { ActionMailer::Base.deliveries.length }.by(1)
```

Comment:
Same comment about checking for enqueued jobs instead of just removing this check.